server/src/main/java/org/elasticsearch/index/stats/IndexingPressureStats.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.stats;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public class IndexingPressureStats implements Writeable, ToXContentFragment {
```

Comment:
You've suggested in Slack doing different buckets (coordinating vs. primary) in metrics, which I think is a good idea, as it will make it clearer that some nodes might only be doing coordination, while others might not be client-facing and therefore doing purely primary (+ replica) work.

Reply:
I explored this on Thursday and ran into some issues.

Essentially it is reasonably straightforward to mark the bytes are coordinating at the `TransportBulkAction` level. But here is the problem I ran into. When a node receives the `TransportShardBulkAction` transport message we mark the bytes. And as we discussed we would mark the bytes as primary in the updated version. However, we use the `rerouteWasLocal` marker to know whether we need to mark the bytes at the `primaryAction` level. In this new version we would need to mark the bytes if the `rerouteWasLocal` but on the `TransportBulkAction` node. But not on the node that receives the `TransportShardBulkAction` message. And I am not sure how to delineate there.

We could mark coordinating bytes when `TransportShardBulkAction` is received. And then mark bytes at the primary level. We can mark a delta when the `rerouteWasLocal` is true. And use that `delta` to avoid double accounted rejections. That just means that a primary node will always mark both coordinating and primary bytes. Technically the reroute component can be coordinating work. But it feels kind of weird.

Let's discuss tomorrow and see if we still want to pursue this breakdown.